edisgo/grid/components.py

```diff
@@ -324,21 +465,87 @@ def timeseries(self):
                     logger.exception("No time series for type {} "
                                      "given.".format(self.type))
                     raise
-            timeseries['q'] = timeseries['p'] * tan(acos(self.power_factor))
+
             timeseries = timeseries * self.nominal_capacity
+            if self.timeseries_reactive is not None:
+                timeseries['q'] = self.timeseries_reactive
+            else:
+                timeseries['q'] = timeseries['p'] * self.q_sign * tan(acos(
+                    self.power_factor))
+
             return timeseries
         else:
             return self._timeseries.loc[
                    self.grid.network.timeseries.timeindex, :]
 
+    @property
+    def timeseries_reactive(self):
+        """
+        Feed-in reactive power time series used to obtain a reactive power time series
+        instead of calculating it from the assumptions
+
+        Returns
+        -------
+        :pandas:`pandas.Series<series>`
+            series containing reactive power in kvar.
+        """
+        if self._timeseries_reactive is None:
+            try:
+                timeseries = \
+                    self.grid.network.timeseries.generation_reactive_power[
+                        self.type].to_frame('q')
+                self.power_factor = None
+                self.reactive_power_mode = None
+                self._timeseries_reactive = timeseries * self.nominal_capacity
+                return self._timeseries_reactive
+            except (KeyError, TypeError):
+                try:
+                    timeseries = \
+                        self.grid.network.timeseries.generation_reactive_power[
+                            'other'].to_frame('q')
+                    self.power_factor = 'not_applicable'
```

Comment:
Why do you set power_factor to 'not_applicable' and in the first try to None?

Reply:
The reason for this is to prevent the timeseries from being accidentally recalculated and changed some value, as if power_factor is set to None then it would re-calculate this from the assumption and get set to 0.9 or 0.95 as in the config file even though a given timeseries_reactive_power exists.